### PR TITLE
Front door ARM Template changes for Security Headers

### PR DESCRIPTION
**Pull request by dborgen2**
## What’s changing?

The Strict-Transport-Security and X-Frame-Options headers were sometimes missing from the query tool and dashboard application due to Front Door/EasyAuth quirks. The easiest and most straightforward solution is to make changes to Front Door itself to pass these header values in the response.

## Why?

Closes NAC-477, NAC-458, and NAC-505

## This PR has:

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

**Comment by edwintorres** [comment]
I'm curious, it was missing on all requests to the applications?

**Comment by dborgen2** [comment]
> I'm curious, it was missing on all requests to the applications?

No, it was very strange. When the user was not logged in, they would be redirected to the login screen but before they were they were given a 200 response. That 200 response didn't have the "preload" portion of the header, and I think was sent by something other than the webapp itself. Perhaps front door or easy auth?

When the user was logged in, the header was identified twice, one with preload and one without. Again, something other than the web app was putting it in there. So I figured just override the value on the front door end, and that seemed to work.

File: iac/arm-templates/front-door-app-service.json [hunk 1]
```diff
@@ -214,19 +214,43 @@
...
                                 }
                             },
+                            "rulesEngine": {
+                                "id": "[resourceId('Microsoft.Network/frontdoors/rulesengines', parameters('frontDoorName'), 'SecurityHeaderRules')]"
+                            },
+                            "enabledState": "Enabled"
+                        }
+                    },
+                    {
+                        "name": "HttpRedirect",
```
Comment (ryangeorgeharvey) [inline iac/arm-templates/front-door-app-service.json:236]: Nitpick (non-blocking): Change name to lower camelCase 'httpRedirect' to match `routingRule1` naming convention.

File: iac/arm-templates/front-door-app-service.json [hunk 2]
```diff
@@ -260,6 +284,40 @@
                     }
                 ]
             }
+        },
+        {
+            "type": "Microsoft.Network/frontdoors/rulesengines",
+            "apiVersion": "2020-05-01",
+            "name": "[concat(parameters('frontDoorName'), '/SecurityHeaderRules')]",
```
Comment (ryangeorgeharvey) [inline iac/arm-templates/front-door-app-service.json:291]: Nitpick (non-blocking): Consider making `SecurityHeaderRules` an ARM variable, as you refer to it multiple places in the ARM.

**Review by ryangeorgeharvey — APPROVED** [review]
Minor code review, otherwise LGTM.

File: iac/arm-templates/front-door-app-service.json (hunk 2 above)
Comment (dborgen2) [reply iac/arm-templates/front-door-app-service.json]: Yea, good idea. Never used ARM template variables before but I think I updated it correctly

File: iac/arm-templates/front-door-app-service.json (hunk 1 above)
Comment (dborgen2) [reply iac/arm-templates/front-door-app-service.json]: 👍

**Pull request merged**